apps/system/js/app_window.js

```diff
@@ -2431,5 +2448,63 @@
       this.statusbar.handleStatusbarTouch(evt, barHeight);
     }
   };
+
+
+  /**
+   * Return a promise that resolves to an icon URL.
+   *
+   * @param {number?} iconSize
+   * @returns {Promise}
+   */
+  AppWindow.prototype.getSiteIconUrl =
+    function ac_getSiteIconUrl(iconSize = ICON_SIZE) {
```

Comment:
My one comment is that I think we might need to take pixel density into account here. Eventually the call to this method will probably want to take an actual argument as it gets used in different places. I think it's fine to just use a default for now, but I think if the default isn't multiplied by pixel density it might end up being blurring on greater than 1x configurations?